lib/Backend/LowerMDSharedSimd128.cpp

```diff
@@ -1327,7 +1336,11 @@ IR::Instr* LowererMD::Simd128LowerShift(IR::Instr *instr)
         IR::RegOpnd * shamtReg = IR::RegOpnd::New(TyInt8, m_func);
         shamtReg->SetReg(LowererMDArch::GetRegShiftCount());
         IR::RegOpnd * tmp = IR::RegOpnd::New(TyInt8, m_func);
-
+#ifdef _M_IX86
```

Comment:
This done elsewhere in LowererMDShared, and in situations where we need the shift amount to be in cl. So it should work here as well (albeit some potential perf loss, if we have to spill ebx).

I haven't looked into why LinearScan doesn't assign a byteable reg. My guess is that a sym will get byteable reg or not upon its def instr. So if we have Ld d(TyInt8), s(TyInt32), d will get a byteable reg, but the problem is that s will not be moved to a byteable reg, if it is not in one. In other words, I don't think value are moved to a byteable reg if a use demands it.